otel-extension/src/main/java/akka/actor/impl/MailboxEnqueueAdvice.java

```diff
@@ -0,0 +1,27 @@
+package akka.actor.impl;
+
+import akka.actor.ActorContext;
+import akka.dispatch.Mailbox;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.instrumentation.api.field.VirtualField;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.Instruments;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.InstrumentsProvider;
+import net.bytebuddy.asm.Advice;
+
+import java.util.Objects;
+
+public class MailboxEnqueueAdvice {
```

Comment:
There's a method in `ActorCell`: 

```
 /**
   * If the actor isLocal, returns the number of "user messages" currently queued,
   * which may be a costly operation, 0 otherwise.
   */
  def numberOfMessages: Int
```

We could (maybe?) pass a reference to that field every time an actor is created. 

```
object LocalActorRefProviderAdvice {

  @OnMethodExit
  def actorOf(@Return ref: ActorRef, @Argument(0) system: ActorSystem): Unit =
    ??? // get the reference, the path and create the instrument for the actorRef (that contains the actorcell, which has the numberOfMessages method).

}
```

that advice could be attached to: 

```
private val localActorRefProviderInstrumentation: AgentInstrumentation =
    instrument("akka.actor.LocalActorRefProvider".fqcnWithTags("create"))
      .visit(LocalActorRefProviderAdvice, "actorOf")
```

(taken from v0.7.0)

everytime we create a new actor maybe we can setup an async instrument that would periodically poll for the value from `numberOfMessages`, thus taking the mailbox size. Gauge seems a good fit. Not sure about the performance of this (probably awful, but maybe not?).


@skipper1982 lmk what do you think. I realize this can be unclear so let's have a call if needed. :)